docs/options.md

```diff
@@ -260,11 +260,66 @@ CIBW_MANYLINUX_X86_64_IMAGE: manylinux2014
 CIBW_MANYLINUX_I686_IMAGE: manylinux2014
 CIBW_SKIP: cp27-manylinux*
 
+# build using the latest manylinux2010 release, instead of the cibuildwheel
+# pinned version
+CIBW_MANYLINUX_X86_64_IMAGE: quay.io/pypa/manylinux2010_x86_64:latest
+CIBW_MANYLINUX_I686_IMAGE: quay.io/pypa/manylinux2010_i686:latest
+
 # build using a different image from the docker registry
 CIBW_MANYLINUX_X86_64_IMAGE: dockcross/manylinux-x64
 CIBW_MANYLINUX_I686_IMAGE: dockcross/manylinux-x86
 ```
 
+### `CIBW_DEPENDENCY_VERSIONS` {: #dependency-versions}
+> Specify how cibuildwheel controls the versions of the tools it uses
+
+Options: `pinned` `latest` `<your constraints file>`
+
+Default: `pinned`
+
+If `CIBW_DEPENDENCY_VERSIONS` is `pinned`, cibuildwheel uses versions of tools
+like `pip`, `setuptools`, `virtualenv` that shipped with that release. This
+represents a known-good set of dependencies, and is recommended for build
+repeatability.
+
+If set to `latest`, cibuildwheel will use the latest of these packages that
+are available on PyPI. This might be preferable if these packages have bug
+fixes that can't wait for a new cibuildwheel release.
+
+To control the versions of dependencies yourself, you can supply a [pip
+constraints](https://pip.pypa.io/en/stable/user_guide/#constraints-files) file
+here and it will be used instead.
+
+!!! note
+    If you need different dependencies for each python version, provide them
+    in the same folder with a `-pythonXY` suffix. e.g. if your
+    `CIBW_DEPENDENCY_VERSIONS=./constraints.txt`, cibuildwheel will use
+    `./constraints-python27.txt` on Python 2.7, or fallback to
+    `./constraints.txt` if that's not found.
+
+Platform-specific variants also available:<br/>
+`CIBW_DEPENDENCY_VERSIONS_MACOS` | `CIBW_DEPENDENCY_VERSIONS_WINDOWS`
+
+!!! note
+    This option does not affect the tools used on the Linux build - those versions
```

Comment:
Hmm. yeah.... we could! Does that defeat the purpose of the manylinux images a little bit? 🤔  

Pinning the manylinux version makes it deterministic already, so this decision doesn't affect that.

Pros:
- more consistent builds between mac/win/linux
Cons:
- longer build time installing deps for each python environment

Oh, actually, there was a problem with the auditwheel constraint in constraints.txt, what was it... ah yes.

Auditwheel depends on [a pinned version of wheel](https://github.com/pypa/auditwheel/blob/master/requirements.txt), so if we include that in constraints.txt we can't update wheel independently.

See this commit https://github.com/joerick/cibuildwheel/pull/256/commits/a953c5f4ee1afe3d16a1407a9c0a5ce69096008d, where removing the auditwheel constraint allowed us to update to the latest wheel version.

Which is.... annoying, to be honest. This indicates that perhaps per-platform constraint file support might be a good idea. But it also says that we probably won't be able to provide cross-platform consistency unless auditwheel was installed using something else, like [pipx](https://github.com/pipxproject/pipx)

Reply:
I would rather use same version of packages like pinned manylinux image than upgrading core packages in it.

Reply:
Yeah, I know, it's not a straightforward discussion/decision, but that's why I wanted to at least mention it. I agree that this is the point of manylinux images, but at the same time, `cibuildwheel` tries to provide this interface to the user that's as much the same for different platforms.

> I would rather use same version of packages like pinned manylinux image than upgrading core packages in it.

@Czaki Why?

> Auditwheel depends on [a pinned version of wheel](https://github.com/pypa/auditwheel/blob/master/requirements.txt), so if we include that in constraints.txt we can't update wheel independently.

Argh, yes, I bumped into that before. Is there a good reason we can't pin `auditwheel` as well? (Oh, yes, I can answer my own question: we're not installing or upgrading auditwheel, so constraint files won't have an effect.)

Maybe that's a good argument for not updating things, indeed :-(

Reply:
>>   I would rather use same version of packages like pinned manylinux image than upgrading core packages in it.

> @Czaki Why?

Because of audithweel which force version of wheel.